### PR TITLE
control report with date range logic update

### PR DESCRIPTION
**Pull request by iamazzeez**
Signed-off-by: Abdul-Az <aazeez@progress.com>
https://chefio.atlassian.net/browse/KINETICS-237

### :nut_and_bolt: Description: What code changed, and why?
If date range filter is passed GetStatsSummaryControls will extract the data from control index else it will behave like old way.

### :athletic_shoe: How to Build and Test the Change
rebuild components/compliance-service

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [x] Tests added/updated? (all new code needs new tests)
- [x] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
Screenshot 2022-10-19 at 12 20 15 PM
Screenshot 2022-10-19 at 12 20 56 PM
Screenshot 2022-10-19 at 12 21 16 PM

### CODE REVIEW

File: components/compliance-service/reporting/relaxting/reports.go [hunk 1]
```diff
@@ -2254,6 +2254,28 @@ func validateFiltersTimeRange(endTime string, startTime string) error {
 	return nil
 }
 
+func isDateRange(endTime string, startTime string) (bool, error) {
+	if len(endTime) == 0 || len(startTime) == 0 {
+		return false, nil
+	}
+	eTime, err := time.Parse(layout, endTime)
+	if err != nil {
+		return false, errors.Errorf("cannot parse the time")
```
Comment (daveaugustus) [inline components/compliance-service/reporting/relaxting/reports.go:2263]:
```suggestion
		return false, errors.Errorf("cannot parse the end time")
```

File: components/compliance-service/reporting/relaxting/reports.go [hunk 2]
```diff
@@ -2254,6 +2254,28 @@ func validateFiltersTimeRange(endTime string, startTime string) error {
...
+	}
+	sTime, err := time.Parse(layout, startTime)
+	if err != nil {
+		return false, errors.Errorf("cannot parse the time")
```
Comment (daveaugustus) [inline components/compliance-service/reporting/relaxting/reports.go:2267]:
```suggestion
		return false, errors.Errorf("cannot parse the start time")
```

**Review by daveaugustus — APPROVED** [review]

File: components/compliance-service/reporting/relaxting/reports.go (hunk 1 above)
Comment (iamazzeez) [reply components/compliance-service/reporting/relaxting/reports.go]: updated

File: components/compliance-service/reporting/relaxting/reports.go (hunk 2 above)
Comment (iamazzeez) [reply components/compliance-service/reporting/relaxting/reports.go]: Updated

**Review by YashviJain01 — APPROVED** [review]

**Pull request merged**